apps/crm/crm/src/app/crm/pages/contracts/view/view.component.ts

```diff
@@ -23,12 +24,12 @@ import { createModalData } from '@blockframes/ui/global-modal/global-modal.compo
 })
 export class ContractViewComponent {
 
-  contract$ = this.route.params.pipe(map(r => r.saleId as string))
-    .pipe(
-      switchMap(saleId => this.getSale(saleId)),
-      filter(contract => !!contract),
-      tap(contract => this.statusForm.setValue(contract.status))
-    );
+  contract$ = this.route.params.pipe(
+    map(r => r.contractId as string),
+    switchMap(id => this.getContract(id)),
+    filter(contract => !!contract),
+    tap(contract => this.statusForm.setValue(contract.status))
+  );
```

Comment:
to avoid `as string` you can do

```suggestion
  contract$ = this.route.params.pipe(
    switchMap(({ contractId }: { contractId: string }) => this.getContract(contractId)),
    filter(contract => !!contract),
    tap(contract => this.statusForm.setValue(contract.status))
  );
```